diff --git a/node_modules/wechaty-puppet-xp/dist/cjs/src/puppet-xp.js b/node_modules/wechaty-puppet-xp/dist/cjs/src/puppet-xp.js
index 5cf904f..9aae8bc 100644
--- a/node_modules/wechaty-puppet-xp/dist/cjs/src/puppet-xp.js
+++ b/node_modules/wechaty-puppet-xp/dist/cjs/src/puppet-xp.js
@@ -788,7 +788,8 @@ class PuppetXp extends PUPPET.Puppet {
     }
     async messageSendFile(conversationId, file) {
         // PUPPET.throwUnsupportedError(conversationId, file)
-        const filePath = path_1.default.resolve(file.name);
+        // const filePath = path_1.default.resolve(file.name);
+        const filePath = path_1.default.resolve(`tmp/${file.name}`);
         wechaty_puppet_1.log.verbose('filePath===============', filePath);
         await file.toFile(filePath, true);
         if (file.type === file_box_1.FileBoxType.Url) {
